ticketbuyer/purchase.go

```diff
@@ -81,11 +92,30 @@ out:
 				case <-quit:
 					return
 				}
-				// Purchase tickets for each attached block, not just for the
-				// update to the main chain.  This is probably not optimal but
-				// it matches how dcrticketbuyer worked.
-				for h := v.NewHeight - int32(len(v.AttachedBlocks)) + 1; h <= v.NewHeight; h++ {
-					p.purchase(int64(h))
+
+				// fetch current peer block heights
+				peerInfo, err := p.w.ChainClient().GetPeerInfo()
+				peerBlockHeights := &Int64Slice{}
+				if err != nil {
+					log.Errorf("Failed to get peer info")
+				}
+
+				for _, peer := range peerInfo {
+					*peerBlockHeights = append(*peerBlockHeights, peer.CurrentHeight)
+				}
+
+				// get the median block height and try a ticket purchase
+				sort.Sort(*peerBlockHeights)
+				var medianHeight int64
+				middle := peerBlockHeights.Len() / 2
+				if peerBlockHeights.Len()%2 == 0 {
+					medianHeight = (peerBlockHeights.Get(middle-1) + peerBlockHeights.Get(middle)) / 2
+				} else {
+					medianHeight = peerBlockHeights.Get(middle)
+				}
```

Comment:
Seems like median keeps getting reimplemented.  Would be nice if go had generics.  Anyway, I'd suggest putting a new median function in dcrutil`, but I'm not sure where would be appropriate.